### PR TITLE
Added button to list content view controller to edit list name

### PR DESCRIPTION
**Pull request by biennviquiera**
**Description**
Users can now edit the name of their owned lists using a button on the top right of the navigation bar.

**DEMO**

https://user-images.githubusercontent.com/66394477/183769075-43dbd3b2-9ce9-4210-9dd9-e57902c73db8.mov

### CODE REVIEW

**Comment by biennviquiera** [comment]
**Demo of camera choice**
https://user-images.githubusercontent.com/66394477/183779260-2f165cd0-a86e-40b7-bfb5-eea0c8e58684.mp4

File: Disher/View Controllers/ListContentViewController.m [hunk 1]
```diff
@@ -99,7 +100,7 @@ - (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEd
         }];
     }
 }
-- (void) reloadListRecipes:(id)something completionHandler:(void(^)(NSArray *returnedRecipes))completionHandler{
+- (void)reloadListRecipes:(id)something completionHandler:(void(^)(NSArray *returnedRecipes))completionHandler{
```
Comment (melissa-gu) [inline Disher/View Controllers/ListContentViewController.m:103]: just noticing this `something`. can we rename

**Comment by ecristerna** [comment]
@biennviquiera there seems to be some duplicate changes from the last PR, can we check this to avoid confusion during review, please?

File: Disher/View Controllers/EditListNameViewController.m [hunk 2]
```diff
@@ -39,7 +42,45 @@ - (IBAction)didPressSave:(id)sender {
...
+  [alertController addAction:takePhoto];
+  UIAlertAction *choosePhoto=[UIAlertAction actionWithTitle:@"Select From Photos" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
+      UIImagePickerController *pickerView = [[UIImagePickerController alloc] init];
+      pickerView.allowsEditing = YES;
+      pickerView.delegate = self;
+      [pickerView setSourceType:UIImagePickerControllerSourceTypePhotoLibrary];
+      [self presentViewController:pickerView animated:YES completion:nil];
+      [alertController dismissViewControllerAnimated:YES completion:nil];
+  }];
+  [alertController addAction:choosePhoto];
```
Comment (melissa-gu) [inline Disher/View Controllers/EditListNameViewController.m:68]: util method here would be good

**Comment by biennviquiera** [comment]
> @biennviquiera there seems to be some duplicate changes from the last PR, can we check this to avoid confusion during review, please?

@ecristerna Sorry, I've been creating a new branch every time I make a PR and it seems that the commits are preserved

File: Disher/View Controllers/ListContentViewController.h [hunk 3]
```diff
@@ -12,7 +12,7 @@
 NS_ASSUME_NONNULL_BEGIN
 
 @protocol ListContentDelegate<NSObject>
-- (void) didUpdateName:(NSString *)name;
+- (void) didUpdateName:(NSString *)name withImage:(UIImage *)image;
```
Comment (melissa-gu) [inline Disher/View Controllers/ListContentViewController.h:15]: should we rename this from `didUpdateName` if it's for both the name and image?

**Review by melissa-gu — APPROVED** [review]

File: Disher/View Controllers/ListContentViewController.h (hunk 3 above)
Comment (biennviquiera) [reply Disher/View Controllers/ListContentViewController.h:15]: It takes two parameters so I just named it didUpdateNameWithImage; objective-c is weird

File: Disher/View Controllers/ListContentViewController.m (hunk 1 above)
Comment (biennviquiera) [reply Disher/View Controllers/ListContentViewController.m:103]: solved; didnt know easier syntax when I wrote this

**Pull request merged**